src/client/src/ui/spotTile/components/PriceButton.tsx

```diff
@@ -92,26 +126,40 @@ const getBigFigureDisplay = (bigFigure: number, rawRate: number) =>
 const renderBigFigureDisplay = (bigFigureDisplay: string) =>
   bigFigureDisplay.toString().length === 3 ? `${bigFigureDisplay}0` : bigFigureDisplay
 
+const ExpiredPrice = styled.div`
+  color: ${({ theme }) => theme.template.red.normal};
+  font-size: 10px;
+  text-transform: uppercase;
+`
+
 const PriceButtonComp: React.FC<PriceButtonProps> = ({
   big = 0,
   pip = 0,
   tenth = 0,
   rawRate = 0,
   direction = Direction.Buy,
   handleClick = () => {},
+  rfqState,
   disabled = false,
+  expired = false,
 }) => {
   const bigFigure = getBigFigureDisplay(big, rawRate)
   return (
-    <TradeButton direction={direction} onClick={() => handleClick()} disabled={disabled}>
-      <Price>
+    <TradeButton
+      direction={direction}
+      onClick={handleClick}
+      rfqState={rfqState}
+      disabled={disabled}
+    >
+      <Price disabled={disabled}>
         <BigWrapper>
           <DirectionLabel>{direction.toUpperCase()}</DirectionLabel>
           <Big>{renderBigFigureDisplay(bigFigure)}</Big>
         </BigWrapper>
         <Pip>{renderPips(pip)}</Pip>
         <Tenth>{tenth}</Tenth>
       </Price>
+      {expired && <ExpiredPrice>Expired</ExpiredPrice>}
```

Comment:
Question: Do we need the additional expired prob. Looking at PriceControls it derives expired from RfqState. Could we use RfqState here?

Reply:
So what I did is that I kept the `expired` and removed the rfqState 😄 ... Yes, we could simply use `rfqState` indeed, but we would then couple the "expired" state of this UI component specifically to the RFQ.

So your comment made me think, what if, i.e "tomorrow", ESP (just like RFQ) or another "trading mode" also has a notion of expired prices and announced prices we will not need to refactor this. Simply pass the proper props and that's it.

Just like an input doesn't care why it's disabled, it just needs to know it's disabled, we explicitly pass disabled and/or priceAnnounced to get the proper state.

Reply:
That's perfect. Nice idea :)